### ISSUE TITLE
valid-lang should check that there is actual content

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3465]
For consistency with the [Element with `lang` attribute has valid language tag](https://w3.org/WAI/standards-guidelines/act/rules/de46e4) ACT rule, axe-core should make sure that the `valid-lang` rule does not fail elements that are empty, or that only contain whitespace characters.

Here are some of the ACT example we're failing, when we shouldn't be:

### [Passed example 4](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/de46e4/8376f95166a75a8541217b77ae6a235f1aac6c3d.html)

```html
<article lang="invalid">
  <div lang="en">
    They wandered into a strange Tiki bar on the edge of the small beach town.
  </div>
</article>
```

### [Inapplicable example 3](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/de46e4/a44f5e11d20feec4ae39e2db0336ddef0a8e04ec.html) 

```html
<p lang="hidden">
  <span style="display: none;">
    They wandered into a strange Tiki bar on the edge of the small beach town.
  </span>
</p>
```

### [Inapplicable example 4](https://www.w3.org/WAI/content-assets/wcag-act-rules/testcases/de46e4/471e3f82cdd9122e2886d2d7bcfc8cda1397a51d.html) 

```html
<div lang="invalid">
  <img src="/WAI/content-assets/wcag-act-rules/test-assets/shared/fireworks.jpg" alt="" />
</div>
```

**Comment by MelSumner** [comment]
👋  we used [is-language-code](https://www.npmjs.com/package/is-language-code) when implementing the `--lang` flag in ember-cli. Sharing in case it's useful! Thank you for all you do! 👍

**Comment by WilcoFiers** [comment]
Thanks Melanie! That looks a little too strict for what we want in axe-core though. Plus the size of it is a bit of an issue. Either way, appreciate the suggestion.

Glad you like our work, if you're ever interested to help, get in touch. I hear good things about your work on Ember too!

**Comment by padmavemulapati** [comment]
Validate with the latest develop branch code base, 
test Script: ` <div id="target"><span lang="en"> text </span></div>`
<img width="1038" alt="image" src="https://user-images.githubusercontent.com/47409579/188797174-2b7903c9-6ae8-41d6-8964-e615c6ba244e.png">
